File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/Val.java
package closer.vlllage.com.closer.handler.helpers;

import java.util.Random;

import closer.vlllage.com.closer.pool.PoolMember;

import static com.google.android.gms.common.util.Strings.isEmptyOrWhitespace;

public class Val extends PoolMember {
    public String rndId() {
        Random random = new Random();
        return Long.toString(random.nextLong()) +
                Long.toString(random.nextLong()) +
                Long.toString(random.nextLong());
    }

    public String of(String string) {
        return string == null ? "" : string.trim();
    }

    public String of(String string, String stringWhenEmpty) {
        return isEmpty(string) ? stringWhenEmpty : string;
    }

    public boolean isEmpty(String string) {
        return isEmptyOrWhitespace(string);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/Val.kt
package closer.vlllage.com.closer.handler.helpers

import com.queatz.on.On
import java.util.*

class Val constructor(private val on: On) {
    fun rndId(): String {
        val random = Random()
        return random.nextLong().toString() +
                random.nextLong() +
                random.nextLong()
    }

    fun trimmed(string: String?): String {
        return of(string?.trim(), "")
    }

    fun of(string: String?, stringWhenEmpty: String): String {
        return if (string.isNullOrBlank()) stringWhenEmpty else string
    }

    inline fun <reified T : Enum<T>> valueOr(type: String, default: T): T = try {
        java.lang.Enum.valueOf(T::class.java, type)
    } catch (e: IllegalArgumentException) {
        default
    }
}
